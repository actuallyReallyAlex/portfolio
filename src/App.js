import React, { Component } from 'react';
import Header from './components/Header'

class App extends Component {
  render() {
    return (
      <Header />
    );
  }
}

export default App;
