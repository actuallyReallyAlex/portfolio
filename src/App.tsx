import React from "react";
import GitHubIcon from "svg-react-loader?name=GitHubIcon!./icons/github.svg";
import EmailIcon from "svg-react-loader?name=EmailIcon!./icons/email.svg";
import LinkedInIcon from "svg-react-loader?name=LinkedInIcon!./icons/linkedin.svg";
import Header from "./components/Header";
import Bio from "./components/Bio";
import ThemeToggle from "./components/ThemeToggle";

export const App = () => {
  return (
    <>
      <Header />
      <article>
        <Bio />
        <section>
          <main>
            <a href="https://github.com/alexlee-dev/" id="github" target="_blank">
              <GitHubIcon />
            </a>
            <a href="mailto:actuallyreallyalex@gmail.com" id="email">
              <EmailIcon />
            </a>
            <a href="https://www.linkedin.com/in/alexlee-dev/" id="linked-in" target="_blank">
              <LinkedInIcon />
            </a>
          </main>
        </section>
      </article>
      <ThemeToggle />
    </>
  );
};

export default App;
