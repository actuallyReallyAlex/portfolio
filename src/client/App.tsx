import * as React from "react";

import { PortfolioItem } from "./types";

export interface AppProps {}

const App: React.FunctionComponent<AppProps> = () => {
  const intialPortfolioItems: PortfolioItem[] = [];
  const [portfolioItems, setPortfolioItems] = React.useState(
    intialPortfolioItems
  );

  React.useEffect(() => {
    const getPortfolioItems = async (): Promise<
      PortfolioItem[] | { error: any }
    > => {
      try {
        const response = await fetch("/portfolioItems", {
          headers: {
            "Content-Type": "application/json",
          },
        });
        const data = await response.json();

        setPortfolioItems(data);
      } catch (error) {
        console.error(error);
        return { error };
      }
    };
    getPortfolioItems();
  }, []);
  return (
    <div>
      <h1>Hello, World!</h1>
      <ul>
        {portfolioItems.map((item) => (
          <li key={item.title}>{item.title}</li>
        ))}
      </ul>
    </div>
  );
};

export default App;
