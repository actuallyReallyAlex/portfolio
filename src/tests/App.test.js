import React from 'react'
import App from '../App'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<App />', () => {
  test('Should render the <App />.', () => {
    const component = shallow(<App {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
