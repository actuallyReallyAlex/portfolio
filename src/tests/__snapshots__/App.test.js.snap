// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render the <App />. 1`] = `ShallowWrapper {}`;
